sdk/core/azure-core/azure/core/pipeline/policies/_retry.py

```diff
@@ -377,7 +385,7 @@ def update_context(self, context, retry_settings):
         if retry_settings['history']:
             context['history'] = retry_settings['history']
 
-    def send(self, request):
+    def send(self, request):    # pylint: disable=too-many-statements, too-many-branches
```

Comment:
When I see this pylint warning, and the fact that we duplicated a ton of code from sync/async, I feel we should have a common logic imported from both places instead.
TLDR: Too much duplication of algorithm/logic here to me